### ISSUE TITLE
Python3 can't convert branches with multiple leafs, Python2 seems to work

### ISSUE THREAD

**Issue by kdbartlett** [scikit-hep/uproot#19]
uproot issue with Python3 when converting rootfile with branches containing multiple leafs. Rootfile has the following leaf structure for a given branch in the tree: "valueX/D:valueY/D:valueZ/D:valueN/I".  

Using the following code to open the rootfile and access branches:
```
import numpy
import uproot

tfile = uproot.open('rootfile_name.root')
ttree = tfile.get('tree')
branch = ttree.get('branch_name')
branch.array()
```

A `TypeError: data type not understood` is raised from the numpy.dtype function on line 141 of /uproot/interp/auto.py when using Python3. 

https://github.com/scikit-hep/uproot/blob/597a2bb03a9ffb45a32e3208be0aed4781986f9f/uproot/interp/auto.py#L141 

I believe this is due to the numpy.dtype function not being able to parse the names of leaves (valueX, valueY, valueZ, valueN) which carry a "bytes" type in python 3 versus a "string" type with Python2. When loading the roofiles into python with uproot are the names of the objects in the tree suppose to be interpreted as "bytes" or "string" types? numpy.dtype seems to have trouble with this depending on which version of python is used.

(Comparing Python versions 3.5.4 and 2.7.13 using numpy version 1.11.2)

**Comment by jpivarski** [comment]
Thanks for catching this! I know I pointed you here from rootconverter, but I'm still in the process of getting uproot ready for a 2.0 release announcement. Enough had been developed for me to put the new code in the master branch, but not enough for me to cut a fixed release or announce it widely.

So it's great to hear that multiple leaves per branch worked in Python 2! You've tested it before me!  :-7

This bytes fix is a simple fix and I just put it in. (Too early for a fixed release also means too early for branch-pull request formality: you get the correction right away.) I just noticed that GitHub closed the issue as soon as I committed it— I'm going to have to hurry up with this explanatory comment.

Unfortunately, I expect to find more Python 3 string vs bytes issues in any code without test coverage. Strings lifted from ROOT files really must be bytes, since ROOT has no intrinsic encoding, so making them Python `bytes` is truth in advertising. However, it's hard to catch all the places where Python 3 wants an `str`. Only code coverage will do that, and it takes time to cover everything. Getting there!

Let me know as soon as you find any `str` vs `bytes` issues— they're easy to fix, so I should stop what I'm doing to fix them one by one. If I had time, I'd be taking all your test cases as unit tests to provide that coverage, but there's the problem that they depend on ROOT files, which might be large, private, etc. I had to reproduce them with my own test files. (None of which have multiple leaves per branch yet, I'm afraid.)
